### ISSUE TITLE
Trailing slash handling in computeRelativePath.

### ISSUE THREAD

**Issue by mgautierfr** [kiwix/kiwix-lib#457]
Problem raised by @veloman-yunkan in PR openzim/zim-tools#229 (https://github.com/openzim/zim-tools/pull/229#issuecomment-788814391)

---

>> P.S. Does computeRelativePath() in libkiwix have to deal with that peculiarity related to the trailing slash?
> This already deal with it :
> -  We already split the paths explicitly keeping "empty" part. So we trailing slash will generate a empty part at the end.
> -  This is tested here : https://github.com/kiwix/kiwix-lib/blob/master/test/pathTools.cpp#L145-L146, https://github.com/kiwix/kiwix-lib/blob/master/test/pathTools.cpp#L149-L150 (although tests may not be exhaustive)

Those tests demonstrate that the trailing slash is handled properly but the absence of a trailing slash isn't (as long as URLs are concerned).

For example

```
  ASSERT_EQ(computeRelativePath(A2("a","b"), A2("a","foo")),
            P2("..","foo"));
  ASSERT_EQ(computeRelativePath(A3("a","b",""), A2("a","foo")),
            P2("..","foo"));
```

which is equivalent in Unix/URL style to

```
  ASSERT_EQ(computeRelativePath("a/b",  "a/foo"), "../foo"); // !!! wrong, should be "./foo" or "foo"
  ASSERT_EQ(computeRelativePath("a/b/", "a/foo"), "../foo"); // correct
```

**Comment by mgautierfr** [comment]
You are right. Although this function is used only once to compute the relative link when generated library xml. It is always used (for now) with the first argument always being a directory.

**Comment by kelson42** [comment]
@mgautierfr @veloman-yunkan It seems, because there is no documentation to that function, that the first argument has to be a directory path? Otherwise, to me, ending the path with a trailing space would to me say this is a directory and otherwise this is a file!
